File: rtiow/rust/main02.rs
fn main() {
    let nx = 200;
    let ny = 100;
    println!("P3");
    println!("{} {}", nx, ny);
    println!("255");
    for j in (0..ny).rev() {
        for i in 0..nx {
            let r = i as f32 / nx as f32;
            let g = j as f32 / ny as f32;
            let b = 0.2;
            let ir = (255.99*r) as i32;
            let ig = (255.99*g) as i32;
            let ib = (255.99*b) as i32;
            println!("{} {} {}", ir, ig, ib);
        }
    }
}

File: rtiow/zig/main02.zig
const std = @import("std");

pub fn main() !void {
const stdout = std.io.getStdOut().outStream();
const nx = 200;
const ny = 100;
try stdout.print("P3\n", .{});
try stdout.print("{} {}\n", .{nx, ny});
try stdout.print("{}\n", .{255});
var j: i32 = ny - 1;
while (j >= 0) {
var i: i32 = 0;
while (i < nx) {
var r: f32 = @intToFloat(i) / nx;
try stdout.print("{}\n", .{6});
i += 1;
}
j -= 1;
}
}
